src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2051,12 +2053,29 @@ export class ResultsViewPageStore {
     });
 
     readonly genes = remoteData<Gene[]>({
-        invoke: async () => fetchGenes(this.hugoGeneSymbols),
+        invoke: async () => {
+
+            const genes = await fetchGenes(this.hugoGeneSymbols);
+
+            if (_.isEqual(this.hugoGeneSymbols, genes.map((gene)=>gene.hugoGeneSymbol) )) {
+                return genes;
+            } else {
+                throw(new Error(ErrorMessages.InvalidGenes));
+            }
+        },
         onResult:(genes:Gene[])=>{
             this.geneCache.addData(genes);
+        },
+        onError:(err)=>{
+            // throwing this allows sentry to report it
+            throw(err);
         }
     });
 
+    @computed get genesInvalid(){
```

Comment:
should we also catch invalid profiles/studies/overall query, like `QueryStore.submitError`?

Reply:
lets discuss at standup.  i happen to know from sentry that we often get invalid genes.  not sure if that's true for those other params.